### ISSUE TITLE
[Bug] Java server is reported using last jar on the classpath

### ISSUE THREAD

**Issue by nicolasff** [raskrebs/sonar#39]
## Describe the bug

Hello!

I run a number of Java processes locally, each serving requests on a different port. I ran `sonar list` and they appeared with the last `.jar` from their command line as the value for their `PROCESS` column; this is not an appropriate or descriptive representation of what each process is running. It would probably make more sense to use the Java class name used as the server's entry point if there is one, or maybe just `java` otherwise.

A simple command line to start a Java process can look like this:
```shell
/Library/Java/JavaVirtualMachines/jdk-21.jdk/Contents/Home/bin/java \
  -classpath /path/to/dep1.jar:/path/to/dep2.jar:/path/to/dep3.jar \
  com.myapp.SomeClass
```

1. The files listed after `-classpath` are the dependencies (libraries), they are separated using colons just like a `PATH` variable.
2. `com.myapp.SomeClass` is the name of a Java class embedded in one of the jars that are on the classpath; this class contains a `main` method, which is the program's entry point.

The issue: when such a program is running, `sonar list` will show its `PROCESS` as the local directory name followed by the `basename` of the last jar listed on the classpath. So in the example above the `PROCESS` column for this server could be something like `my-app/dep3.jar`. Often, this jar is just a random dependency, it doesn't really matter that it appears last. It could be a logger, some helper library, anything really. It doesn't tell us anything about what the server is actually doing, whereas the name of the class used as the program's entry point will almost always be more informative.

I noticed this issue after running `sonar list` on a machine that runs multiple Java servers that all have a similar classpath, but each one is started using a different class as their entry point. In this output, the `PROCESS` column for each one of those is often the same, and is always irrelevant:

<p align=center>
<img width="600" src="https://github.com/user-attachments/assets/1b22e7a7-84d8-41c4-b8c6-faa10e7edf36" />
</p>

None of the jars listed here have anything to do with these services, they're just random dependencies that happened to be listed last on the classpath. To give you an idea, I picked the first one on port `7003` and split the `-classpath` value on `:`… this one parameter references 412 separate jars, all joined together by colons.

Please note that it is entirely possible for the command not to include a class name and for this class name to be resolved by the JVM at runtime based on some metadata included in one of the jars. In that case, the best way to get a relevant name for the server is to _probably_ use `jps -lm $pid`, although I'd double-check that. Just using `java` could also work.

## Steps to reproduce

1. Download `build.sh`, `run.sh`, and `Server.java` from [this Gist](https://gist.github.com/nicolasff/36c02912b65077db8cf1167b75ab525d). This is a Java server that opens a port and does nothing else. The two jar files that it adds to its class path are related to Java logging, but they are not actually used by the program itself. SLF4J is an abstraction layer for various logging frameworks in Java.
2. I see that Claude Code helped build this repo, so **please** have it or any other agent review these files before running anything. I've tried to keep them as simple as possible to make sure everything can be easily understood.
3. Run `./build.sh` to download the two jar files into a `libs/` directory.
4. Run `./run.sh` to start the Java server with these two jar files on its classpath.
5. Run `sonar list` and find the entry for port `9999`: the `PROCESS` column should say `java-server/slf4j-simple-2.0.9.jar` if you copied these files in a directory named `java-server`.

## Expected behavior

I would have expected to see the Java class name (`Server`) used as the entry point for this Java server, something like this:
```
$ sonar list | grep -E 'PORT|9999'
PORT    PROCESS       CONTAINER     IMAGE             CPORT   URL                   
9999    Server                                                http://localhost:9999 
```

Or showing `java` could also work, but would be less informative.

## Actual behavior

An unrelated dependency is partially mentioned as the `PROCESS` name. One jar file's `basename` is certainly not representative of what this server process is doing, or what it's running. 

With the repro linked above, started from a directory named `java-server`, it will look like this:

```none
$ sonar list | grep -E 'PORT|9999'
PORT    PROCESS                                CONTAINER     IMAGE       CPORT   URL                   
9999    java-server/slf4j-simple-2.0.9.jar                                       http://localhost:9999 
```

## Environment
- OS: macOS Tahoe 26.2 (build `25C56`)
- sonar version: latest `main`, commit 02994f6.
